cleaner/scripts/reclaimable_docker_cache.sh

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#!/bin/bash
+
+docker system df | \
+gawk 'BEGIN {s=0}
+      END {print s}
+      match($5, /([0-9]+\.?[0-9]*)/, a) {if(a[2] == "M") s += a[1]/1024; else s += a[1]}'
```

Comment:
>match($5, /([0-9]+\.?[0-9]*)/, a) {if(a[2] == "M") s += a[1]/1024; else s += a[1]}' [](start = 6, length = 83)

Can you explain a little bit of the rule a little bit here? For example to give some examples?

Reply:
I saw GB also there, does your code handle it?

---
In reply to: [215850149](https://github.com/Microsoft/pai/pull/1292#discussion_r215850149) [](ancestors = 215850149)